modules/nlu/src/views/full/common/TrainNow.tsx

```diff
@@ -37,9 +39,13 @@ const TrainNow: FC<{ api: NLUApi; eventBus: any }> = ({ api, eventBus }) => {
     }
   }
 
+  const renderTrain = () => {
+    return autoTrain ? lang.tr('module.nlu.retrainAll') : lang.tr('module.nlu.trainNow')
+  }
+
```

Comment:
No need to use a method for that, the whole function is re-evaluated everytime it's rendered. `const trainLabel = autoTrain ? ...`

Reply:
What do you think about this instead:

```
  const RenderText = () => {
    let buttonMsg = ""
    if (training) {
      buttonMsg = lang.tr('module.nlu.cancelTraining')
    } else if (autoTrain) {
      buttonMsg = lang.tr('module.nlu.retrainAll')
    } else {
      buttonMsg = lang.tr('module.nlu.trainNow')
    }
    return <div>{buttonMsg}</div>
  }

  return (
    <Button loading={loading} onClick={onClick}>
      <RenderText />
    </Button>
  )
```

Reply:
I usually prefer to avoid render() methods since we got rid of classes (maybe just a personal preference).

What do you think of this approach?
```js
if(training){
  return (
    <Button loading={loading} onClick={cancelTraining}>
      {lang.tr('module.nlu.cancelTraining')}
    </Button>
  )
}

return (
    <Button loading={loading} onClick={trainNow}>
      {lang.tr(autoTrain ? 'module.nlu.retrainAll' : 'module.nlu.trainNow')}
    </Button>
  )
```

Reply:
:+1: